src/main/java/htsjdk/samtools/util/IntervalList.java

```diff
@@ -379,13 +379,14 @@ public void unique(final boolean concatenateNames) {
     }
 
     /**
-     * Merges a sorted collection of intervals and optionally concatenates unique names or takes the first name.
+     * Merges a collection of intervals and optionally concatenates unique names or takes the first name.
+     * Uses the contig and strand of the first interval seen without validating that the others match.
      */
     static Interval merge(final Iterable<Interval> intervals, final boolean concatenateNames) {
         final Interval first = intervals.iterator().next();
         final String chrom = first.getContig();
         int start = first.getStart();
-        int end = start;
+        int end = first.getEnd();
```

Comment:
what should happen when we merge a zero-length interval with a non-zerolength one after it?

so, 
10-9 merged with 11-15  should remain those two intervals
10-9 merged with 10-15 should become one interval 10-15
10-9 merged with 9-15 should become one interval 9-15

Reply:
The first case, you mean it should become 10-15?  Merge never leaves them separate.

Reply:
no...I think that it should not become 10-15....since the size of the first interval is 0 it shouldn't change the total number of bases covered...

Reply:
I think that zero length intervals need to either be dropped when merging, or ignored...normal rules of merging do not apply.